# Piping Calculations

Import the fluids library.  Check the readme file here: http://fluids.readthedocs.io/index.html

In [1]:
import fluids

Given:   Coke oven gas having a specific gravity of 0.42, a header pressure of 125 psig, and a temperature of 140 degF is flowing through 20 feet of 3" Schedule 40 pipe before discharging to atmosphere.  Assume raio of specific heats, k=1.4


Find:  The flow rate in standard cubic feet per hour (scfh)

Get fluid property for water

# Solution based on Thermo module documentation

In [2]:
from fluids.units import *
import numpy as np
from math import sqrt

In [3]:
L = 20*u.feet

Sg = 0.42

P1 = 125*u.psi+14.7*u.psi

T = (140*u.degF)

d = nearest_pipe(NPS=3*u.inch, schedule='40')[1] #inner pipe dimension

D = d.to(u.feet)

D, d, T, P1, L

(<Quantity(0.2556430446194226, 'foot')>,
 <Quantity(0.07792, 'meter')>,
 <Quantity(140, 'degF')>,
 <Quantity(139.7, 'pound_force_per_square_inch')>,
 <Quantity(20, 'foot')>)

Find K

In [4]:
#for pipe
f = 0.0175 #Reading off the Moody diagram, Crane 410, A-26

K_pipe = f*L/D

#for entrance
K_entrance = 0.5 #Crane 410, A-30

#for exit
K_exit = 1.0 #Crane 410, A-30

Ktot = np.sum([K_pipe, K_entrance, K_exit])

Ktot

2.8690965092402463

Find deltaP/P1

In [5]:
deltaP_P1 = (P1 - 14.7*u.psi)/P1

deltaP_P1

<Quantity(0.8947745168217609, 'dimensionless')>

Using the chart on Crane 410 page A-23 for k=1.4, it is found hat for K=2.87, the maximum deltaP/P1 is 0.657 (interpolated from the table on page A-23).  Since deltaP/P1 is less than indicated (deltaP/P1 above of 0.89), sonic velocity occurs at the end of the pipe and we need to calculate a deltaP for further calculations.

In [6]:
delta_P = 0.657*P1

delta_P

<Quantity(91.7829, 'pound_force_per_square_inch')>

In [7]:
T = T.to(u.degR)

T

<Quantity(599.6699996, 'degR')>

Find net expansion factor for compressible flow

In [8]:
Y = 0.637 #Interpolated from Crane 410, A-23

Find the flow rate

In [9]:
qh = (40700*Y*(d.to(u.inch)**2)*sqrt((delta_P*P1/(Ktot*T*Sg)).magnitude)).magnitude*u.feet**3/u.hour

'{:.2f}'.format(qh)

'1027755.96 foot ** 3 / hour'